src/v3/src/util/idxUtils.ts

```diff
@@ -62,6 +63,26 @@ export const getUserInfo = (transaction: IdxTransaction): UserInfo => {
   return user.value as UserInfo;
 };
 
+export const getUserProvidedUserInfo = (data: FormBag['data']): UserInfo => {
+  const identifier = ('userProfile.login' in data
+    ? data['userProfile.login']
+    : data['userProfile.email']) as string;
+  const firstName = 'userProfile.firstName' in data
+    ? data['userProfile.firstName'] as string
+    : undefined;
+  const lastName = 'userProfile.lastName' in data
+    ? data['userProfile.lastName'] as string
+    : undefined;
+
+  return {
+    identifier,
+    profile: {
+      firstName,
+      lastName,
+    },
+  };
+};
+
```

Comment:
Looks awkward. Does it make sense to expand the dot-separated key/value pairs of `userProfile` into an object? Why is it stored this way in the `FormBag`?

Reply:
The reason its stored this way is because of how the widget interfaces with the IDX payload. The object is [flattened](https://github.com/okta/okta-signin-widget/blob/master/src/v3/src/transformer/field/transform.ts#L141) (assuming it make it easier to work with - but also mimics how Gen 2 handles form fields) into simple key/value pairs, with the key being the flattened reference to the data field. 

Keep in mind, every keystroke will update this field value entry in the form bag, so expanding it here doesn't provide much value.